projects/components/src/modal/modal-container.component.ts

```diff
@@ -0,0 +1,83 @@
+import { ChangeDetectionStrategy, Component, Inject, Injector, TemplateRef, Type } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { LayoutChangeService } from '@hypertrace/common';
+import { ButtonSize, ButtonStyle } from '../button/button';
+import { POPOVER_DATA } from '../popover/popover';
+import { PopoverRef } from '../popover/popover-ref';
+import { ModalConfig, ModalSize, MODAL_DATA } from './modal';
+
+@Component({
+  selector: 'ht-modal-container',
+  styleUrls: ['./modal-container.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div *ngIf="this.visible" class="modal-container" [ngClass]="'modal-size-' + this.size">
+      <div *ngIf="this.showHeader" class="header">
+        <ht-button
+          class="close-button"
+          icon="${IconType.Close}"
+          display="${ButtonStyle.Outlined}"
+          size="${ButtonSize.ExtraSmall}"
+          htTooltip="Close modal"
+          (click)="this.close()"
+        >
+        </ht-button>
+        <h3 class="header-title">{{ modalTitle }}</h3>
```

Comment:
Is there a reason to use h3 if we just style it ourselves anyway? Do we have global styles for h# elements?

Reply:
Semantics mainly. We should always be using the most appropriate tags given the content they contain to better mark up the meaning of the document, the styling can still be done by classes. That said, this is pre-existing from the previous implementation: https://github.com/hypertrace/hypertrace-ui/blob/60d954c5b5e7991ce106870f163b49dd85fbea94/projects/components/src/overlay/modal/modal-overlay.component.ts#L24